### PR TITLE
WT-10707 Check return values in push_pull.c

### PR DESCRIPTION
**Pull request by mariammojid**
Coverity reporting values for unchecked return values.

Add `testutil` checks to return values.

### CODE REVIEW

File: bench/tiered/push_pull.c [hunk 1]
```diff
@@ -254,11 +254,11 @@ remove_local_cached_files(const char *home)
 
         ++nmatches;
 
-        sscanf(dir_entry->d_name, "%*[^0-9]%d", &objnum);
+        testutil_assert(sscanf(dir_entry->d_name, "%*[^0-9]%d", &objnum) == 1);
         highest = WT_MAX(highest, objnum);
     }
 
-    closedir(dir);
+    testutil_check(closedir(dir));
```
Comment (marcbutler) [inline bench/tiered/push_pull.c:261]: I feel like we have an opportunity here to enhance the `testutil_` macros. Specifically in the case where the API call returns a negative value (typically -1) and sets `errno`. For such API calls, the specific error code and therefore context will not be available.

The `closedir()` calls is not a good example of this as there is only one errno value (on Linux `EBADF`); but look at the man page  for `close()`.

Comment (mariammojid) [reply bench/tiered/push_pull.c]: Thank you for the suggestion! I went with `testutil_assert_errno`

**Review by ajmorton — APPROVED** [review]
LGTM 👍

**Review by marcbutler — APPROVED** [review]
LGTM 👍

**Comment by mariammojid** [comment]
evergreen merge

**Comment by mariammojid** [comment]
evergreen merge

**Pull request merged**